File: exercise/learnyouakotlin/part2/Sessions.java
package learnyouakotlin.part2;

import learnyouakotlin.part1.Session;

import javax.annotation.Nullable;
import java.util.List;

class Sessions {

    public static @Nullable String subtitleOf(@Nullable Session session) {
        if (session == null)
            return null;
        else
            return session.getSubtitle();
    }

    public static String subtitleOrPrompt(Session session) {
        if (session.getSubtitle() == null)
            return "click to enter subtitle";
        else
            return session.getSubtitle();
    }

    public static @Nullable Session findWithTitle(List<Session> sessions, String title) {
        return sessions.stream().filter(session -> session.getTitle().equalsIgnoreCase(title)).findFirst().orElse(null);
    }

}

File: exercise/learnyouakotlin/part2/Sessions.kt
package learnyouakotlin.part2

import learnyouakotlin.part1.Session

val Session?.subtitle: String?
    get() = this?.subtitle

fun Session.subtitleOrPrompt(): String = subtitle ?: "click to enter subtitle"

fun findWithTitle(sessions: List<Session>, title: String): Session? =
    sessions.find { it.title.equals(title, ignoreCase = true) }
